### ISSUE TITLE
DuckDB /Postgres Dialects should use TRY_CAST for the Malloy :: operator.

### ISSUE THREAD

**Issue by lloydtabb** [malloydata/malloy#1163]
:: means safe_cast in BigQuery, It should mean TRY_CAST in the other dialects.

**Comment by lloydtabb** [comment]
Looks like there is no 'TRY_CAST' for Postgres.  There are some ugly workarounds...  I'm going to fix for duckdb and leave postgres for now...

https://stackoverflow.com/questions/10306830/postgres-define-a-default-value-for-cast-failures

**Comment by lloydtabb** [comment]
Love to have your thoughts on this..  This is one of those lowest common denominator problems...

**Comment by lloydtabb** [comment]
I actually really love `::` for cast  perhaps a separate safe case operater `:::`

```
project: 
  num is num::number  // generates CAST(num AS float68) as num
  safe_num is num:::number // generates SAFE_CAST(num AS float68) as num
```

and posgres doen't support the `:::` operator

**Comment by lloydtabb** [comment]
@mtoy-googly-moogly ^^ what do you think?

**Comment by lloydtabb** [comment]
BTW, generally safe_cast is used during the ETL process when cleaning up data.   You really want to use normal cast when doing table to table stuff.  SAFE_CAST is the exception, generally.

**Comment by carlineng** [comment]
`:::` as "safe_cast" makes sense to me. I think it's a common enough pattern that it deserves its own operator.

**Comment by mtoy-googly-moogly** [comment]
Chris is correct, malloy has two cast gestures ... `::` and `cast` and `::` generates a safe cast on bigquery. I think the change which should go in immediately is to use try_cast on `::` in duckdb

We should have a group chat about `:::` vs. `cast(x as `

**Comment by mtoy-googly-moogly** [comment]
I think the way it is now is that `::` is cast and `:::` is safe cast, for all dialects. I'll let @christopherswenson comment, but I think we should close this?

**Comment by christopherswenson** [comment]
I think this needs a @lloydtabb look to see if he thinks it's correct. I think it's correct now?

**Comment by lloydtabb** [comment]
> I think the way it is now is that `::` is cast and `:::` is safe cast, for all dialects. I'll let @christopherswenson comment, but I think we should close this?

Yes, close.  This is correct on dialects that support safe cast (postgres doesn't)
